tests/probes_test.go

```diff
@@ -69,22 +60,44 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 			return vmiReady(readVmi) == v1.ConditionTrue
 		}
 
-		table.DescribeTable("should succeed", func(readinessProbe *v12.Probe, serverStarter func(vmi *v12.VirtualMachineInstance, port int)) {
-			By("Specifying a VMI with a readiness probe")
-			vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
+		table.DescribeTable("should succeed", func(readinessProbe *v12.Probe, IPFamily v1.IPFamily) {
+
+			if IPFamily == v1.IPv6Protocol {
+				libnet.SkipWhenNotDualStackCluster(virtClient)
+				By("Create a support pod which will reply to kubelet's probes ...")
+				probeBackendPod, supportPodCleanupFunc := buildProbeBackendPodSpec(readinessProbe)
+				defer func() {
+					Expect(supportPodCleanupFunc()).To(Succeed(), "The support pod responding to the probes should be cleaned-up at test tear-down.")
+				}()
+
+				By("Attaching the readiness probe to an external pod server")
+				supportPodIP := libnet.GetPodIpByFamily(probeBackendPod, IPFamily)
+				Expect(supportPodIP).ToNot(BeEmpty())
+				readinessProbe = patchProbeWithIPAddr(readinessProbe, supportPodIP)
```

Comment:
You can encapsulate these in an helper function. Hopefully that will improve readability and remove the need for the `By` clause.

Reply:
Done.